packages/math-new/src/index.ts

```diff
@@ -0,0 +1,239 @@
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
+    return this._tokenInForExactSpotPriceAfterSwap({
+      spotPrice: bnum(spotPrice),
+      poolPairData,
+    }).toNumber();
+  }
+
+  tokenOutForExactSpotPriceAfterSwap(
+    spotPriceAfterSwap: number,
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const reversedPoolPairData = this.math.parsePoolPairData(tokenOut, tokenIn);
+    const amountTokenIn = this._tokenInForExactSpotPriceAfterSwap({
+      spotPrice: bnum(spotPriceAfterSwap),
+      poolPairData: reversedPoolPairData,
+    }).toNumber();
+    return this.exactTokenInForTokenOut(amountTokenIn, tokenIn, tokenOut);
+  }
+
+  effectivePriceForExactTokenInSwap(
+    amountIn: number,
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
+    const amountOut = this.math._exactTokenInForTokenOut(
+      poolPairData,
+      bnum(amountIn)
+    );
+    return amountOut.div(amountIn).toNumber();
+  }
+
+  effectivePriceForExactTokenOutSwap(
+    amountOut: number,
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
+    const amountIn = this.math._tokenInForExactTokenOut(
+      poolPairData,
+      bnum(amountOut)
+    );
+    return bnum(amountOut).div(amountIn).toNumber();
+  }
+
+  priceImpactForExactTokenInSwap(
+    amountIn: number,
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
+    const amountOut = this.math._exactTokenInForTokenOut(
+      poolPairData,
+      bnum(amountIn)
+    );
+    const effectivePrice = amountOut.div(amountIn);
+    const spotPrice = this.math._spotPrice(poolPairData);
+    return bnum(1).minus(effectivePrice.div(spotPrice)).toNumber();
+  }
+
+  priceImpactForExactTokenInReversedSwap(
+    amountIn: number,
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
+    const amountOut = this.math._exactTokenInForTokenOut(
+      poolPairData,
+      bnum(amountIn)
+    );
+    const effectivePrice = amountOut.div(amountIn);
+    const spotPrice = this.math._spotPrice(poolPairData);
+    return bnum(1).minus(spotPrice.div(effectivePrice)).toNumber();
+  }
+
+  priceImpactForExactTokenOutSwap(
+    amountOut: number,
+    tokenIn: string,
+    tokenOut: string
+  ): number {
+    const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
+    const amountIn = this.math._tokenInForExactTokenOut(
+      poolPairData,
+      bnum(amountOut)
+    );
+    const effectivePrice = bnum(amountOut).div(amountIn);
+    const spotPrice = this.math._spotPrice(poolPairData);
+    return bnum(1).minus(effectivePrice.div(spotPrice)).toNumber();
+  }
+
+  priceImpactForExactTokenOutReversedSwap(
```

Comment:
all the AMM should have this function as a common point or this is just the example?